src/popup/pages/Popups/MessageSign.vue

```diff
@@ -46,44 +46,52 @@
   </Modal>
 </template>
 
-<script>
-import { mapState, mapGetters } from 'vuex';
+<script lang="ts">
+import { computed, defineComponent, PropType } from '@vue/composition-api';
+import type { IAccount, IAppData } from '../../../types';
+import { useGetter } from '../../../composables/vuex';
 import Modal from '../../components/Modal.vue';
 import BtnMain from '../../components/buttons/BtnMain.vue';
 import TransactionInfo from '../../components/TransactionInfo.vue';
 import DetailsItem from '../../components/DetailsItem.vue';
 import CopyText from '../../components/CopyText.vue';
-import mixin from './mixin';
 
-export default {
+export default defineComponent({
   components: {
     Modal,
     BtnMain,
     TransactionInfo,
     DetailsItem,
     CopyText,
   },
-  mixins: [mixin],
   props: {
     message: { type: String, required: true },
-    app: { type: Object, required: true },
+    app: { type: Object as PropType<IAppData>, required: true },
+    resolve: { type: Function as PropType<() => void>, required: true },
+    // eslint-disable-next-line no-unused-vars
+    reject: { type: Function as PropType<(e: Error) => void>, required: true },
   },
-  computed: {
-    ...mapGetters([
-      'isConnected',
-      'getExplorerPath',
-    ]),
-    ...mapState({
-      account(_, { account }) {
-        return {
-          ...account,
-          label: this.$t('transaction.overview.accountAddress'),
-          url: this.getExplorerPath(account.address),
-        };
-      },
-    }),
+  setup(props, { root }) {
+    const isConnected = useGetter('isConnected');
+    const getExplorerPath = useGetter('getExplorerPath');
+    const account = useGetter<IAccount>('account');
+    const accountExtended = computed(() => ({
+      ...account.value,
+      label: root.$t('transaction.overview.accountAddress'),
+      url: getExplorerPath.value(account.value.address),
+    }));
```

Comment:
I remember that this exact same object is also used int transaction.ts if fetchOwnershipAccount function - maybe it could be extracted somewhere?

Reply:
I confirm that similar code exists in few places. I tried to create utility function, but besides the repeated `label` there was no common data to return. It was always merging in the account and getting explorer path from provided address. So I decided to use your interface to type those places for now. In the future we could think about something better.